homeassistant/components/keenetic_ndms2/__init__.py

```diff
@@ -79,6 +79,7 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
             if (
                 entity_entry.config_entry_id == config_entry.entry_id
                 and entity_entry.domain == Platform.DEVICE_TRACKER
+                and entity_entry.device_id
```

Comment:
This is required for `async_update_device` below.
But maybe we should do this check after `async_remove`?

Reply:
Yes, it feels safer to not change the entity registry adjustment logic to cater for the device registry state.

Reply:
I've moved it to where is it needed